truth_justice_cake_cutting.py

```diff
@@ -0,0 +1,340 @@
+"""
+
+Programmer: Ofec Israel
+
+"""
+
+import agents
+from agents import PiecewiseLinearAgent
+from agents import PiecewiseUniformAgent
+from allocations import *
+from random import shuffle
+import logging
+logger = logging.getLogger(__name__)
+    
+
+
+
+"""
+Get the intersection of two intervals
+"""
+def intervals_intersection(a,b):
+    intervals = []
+    for i in a:
+        for j in b:
+            interval = (max(i[0], j[0]), min(i[1], j[1]))
+            if(not interval[0]>=interval[1]):
+                intervals.append(interval)
+                
+    return intervals
+
+"""
+Get first element
+"""
+def first_element(i):
+    return i[0]
+
+"""
+Net implementation
+"""
+class Edge(object):
+  def __init__(self, u, v, w):
+    self.source = u
+    self.target = v
+    self.capacity = w
+
+  def __repr__(self):
+    return "%s->%s:%s" % (self.source, self.target, self.capacity)
+
+
+class FlowNetwork(object):
+  def  __init__(self):
+    self.adj = {}
+    self.flow = {}
+
+  def AddVertex(self, vertex):
+    self.adj[vertex] = []
+
+  def GetEdges(self, v):
+    return self.adj[v]
+
+  def AddEdge(self, u, v, w = 0):
+    if u == v:
+      raise ValueError("u == v")
+    edge = Edge(u, v, w)
+    redge = Edge(v, u, 0)
+    edge.redge = redge
+    redge.redge = edge
+    self.adj[u].append(edge)
+    self.adj[v].append(redge)
+    # Intialize all flows to zero
+    self.flow[edge] = 0
+    self.flow[redge] = 0
+    
+  def FindPath(self, source, target, path):
+    if source == target:
```

Comment:
Please either add doctests to each function, or replace these functions with the corresponding functions in networkx.

Reply:
(I think it will be easier for you to use networkx)